oppia/oppia-android
# Fixes part of #40 & #42: Generalisation Highfi Mobile Portrait + Landscape #1616

## Explanation


This PR implements the highfi for DragAndDrop Interaction outer margin only. This PR does not work on the internal design of the interaction.

## How to test:
1. Checkout out the description/references below.
2. In each file I have mentioned margin values in the comments, you can just verify that those margins values are correctly mentioned in code.
3. Run the `Prototype Exploration` to see drag and drop under 6 circumstances:
- Mobile Landscape
- Mobile Portrait
- Tablet Landscape (7 inch or less)
- Tablet Portrait (7 inch or less)
- Tablet Landscape (7 inch or more - Split)
- Tablet Portrait (7 inch or more - Split)

4. Unfortunately we cannot test this in question player because we don't have any question for that.
5. In any xml files if there any changes in the child views of root layout than those changes are just because of reformatting.

## Margin Values
**Mobile-Portrait**
E 40, 24, 24, 0
Q 32, 24, 32, 0
E S  32, 40, 32 ,0
Q S  32, 40, 32 ,0

**Mobile-Landscape**
E 80, 24, 48, 0
Q 64, 24, 64, 0
E S 32, 40, 32, 0
Q S 32, 40, 32, 0

**Tablet Portrait**
E 144, 24, 112, 0
Q 128, 24, 128, 0
E S 32, 40, 32 ,0
Q S  32, 40, 32 ,0

**Tablet-Landscape**
E 208, 24, 176, 0
Q 192, 24, 192, 0
E S 32, 40, 32, 0
Q S 32, 40, 32, 0

## References:

https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/97cfe444-80c8-4d42-be4e-0adaacbaab62/

https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/4d120159-d03f-4f48-82c0-f162565e5745/

https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/f6306f80-9293-4a83-ae34-4aebc4f7346b/

![exploration_landscape](https://user-images.githubusercontent.com/9396084/90193096-e0f8c000-dde1-11ea-9522-1a564eedb152.png)
![exploration_portrait](https://user-images.githubusercontent.com/9396084/90193102-e35b1a00-dde1-11ea-8d02-4f53143451b3.png)
![question_landscape](https://user-images.githubusercontent.com/9396084/90193103-e3f3b080-dde1-11ea-9519-1de869175cbb.png)
![question_portrait](https://user-images.githubusercontent.com/9396084/90193107-e524dd80-dde1-11ea-9a59-27bd1d9567d8.png)


## Screenshots
![Screenshot_1597357160](https://user-images.githubusercontent.com/9396084/90193179-100f3180-dde2-11ea-9e44-142c36ca275f.png)
![Screenshot_1597357166](https://user-images.githubusercontent.com/9396084/90193185-13a2b880-dde2-11ea-9991-3c6bd05c4474.png)
![Screenshot_1597357252](https://user-images.githubusercontent.com/9396084/90193190-14d3e580-dde2-11ea-8dac-24063a106d6c.png)
![Screenshot_1597357257](https://user-images.githubusercontent.com/9396084/90193192-156c7c00-dde2-11ea-8f9d-f224e02ba8bd.png)
![Screenshot_1597357321](https://user-images.githubusercontent.com/9396084/90193193-16051280-dde2-11ea-8303-4c7735b12226.png)
![Screenshot_1597357324](https://user-images.githubusercontent.com/9396084/90193196-169da900-dde2-11ea-8720-28d2c55bb6f9.png)
![Screenshot_1597357333](https://user-images.githubusercontent.com/9396084/90193198-169da900-dde2-11ea-9be1-91bdc808403d.png)


## Checklist
